leasing/management/commands/import_leasehold_transfers.py

```diff
@@ -214,10 +215,16 @@ def _handle_xml_file(self, xml_file):
 
                 institution_identifier = entry.find(".//y:laitostunnus", NS).text
 
-                transfer = LeaseholdTransfer.objects.create(
+                transfer = LeaseholdTransfer.objects.filter(
```

Comment:
On the ticket it was mentioned that one of the problems was that already deleted LeaseholdTransfers appear back. This query probably does not find those, as `LeaseholdTransfer`  model inherits its manager for `objects` from `SafeDeleteModel`. And to my understanding `SafeDeleteManager` does not include objects that have a date set to field `deleted`. I wonder if https://django-safedelete.readthedocs.io/en/latest/managers.html#safedelete.managers.SafeDeleteAllManager could be used here. So `LeaseholdTransfer.all_objects.filter()`.

But after that there is the question what does checking the `.last()` signify? What if this query returns 10 results, does it exist then or not? I am not yet familiar with this concept of LeaseholdTransfer so I can't say.

Perhaps as Juho suggested `get_or_create` could be used here instead, if there is not specific logic with the filter().last()

Reply:
get_or_create breaks when get returns more than 1. In our case there are duplicates though I suppose it could make more sense if leasehold transfers were unique with institution_identifier and decision_date but currently there are already duplicates in database.

The purpose of the `.last()` is just to check that at least 1 leasehold transfer exists (then it exists).

I will give the `all_objects` a try.

Reply:
That makes sense if there are duplicates already.
What should we do to the duplicates in your opinion?

Reply:
I suppose we could try to come up with an sql query to remove duplicates and then we can change the code here to use get_or_create as well. I can make a jira task for this and also write that we can change the code here after duplicates have been removed.